dynawo/sources/Models/CPP/Controls/Voltage/ModelVoltageMeasurementsUtilities/DYNModelVoltageMeasurementsUtilities.cpp

```diff
@@ -0,0 +1,348 @@
+
+using parameters::ParametersSet;
+
+/**
+ * @brief ModelVoltageMeasurementsUtilitiesFactory getter
+ *
+ * @return A pointer to a new instance of ModelVoltageMeasurementsUtilitiesFactory
+ */
+extern "C" DYN::SubModelFactory* getFactory() {
+  return (new DYN::ModelVoltageMeasurementsUtilitiesFactory());
+}
+
+/**
+ * @brief ModelVoltageMeasurementsUtilitiesFactory destroy method
+ */
+extern "C" void deleteFactory(DYN::SubModelFactory* factory) {
+  delete factory;
+}
+
+/**
+ * @brief ModelVoltageMeasurementsUtilities getter
+ *
+ * @return A pointer to a new instance of ModelVoltageMeasurementsUtilities
+ */
+extern "C" DYN::SubModel* DYN::ModelVoltageMeasurementsUtilitiesFactory::create() const {
+  DYN::SubModel* model(new DYN::ModelVoltageMeasurementsUtilities());
+  return model;
+}
+
+/**
+ * @brief ModelVoltageMeasurementsUtilitiesFactory destroy method
+ */
+extern "C" void DYN::ModelVoltageMeasurementsUtilitiesFactory::destroy(DYN::SubModel* model) const {
+  delete model;
+}
+
+namespace DYN {
+
+/**
+ * @brief ModelVoltageMeasurementsUtilities model default constructor
+ *
+ */
+ModelVoltageMeasurementsUtilities::ModelVoltageMeasurementsUtilities() :
+ModelCPP("voltageMeasurementsUtilities"),
+nbConnectedInputs_(0) {
+}
+
+/**
+ * @brief ModelVoltageMeasurementsUtilities model initialization
+ *
+ */
+void
+ModelVoltageMeasurementsUtilities::init(const double /*t0*/) {
+  // Nothing to initalize here.
+}
+
+void
+ModelVoltageMeasurementsUtilities::initializeFromData(const boost::shared_ptr<DataInterface>& /*data*/) {
+  // not needed
+}
+
+/**
+ * @brief ModelVoltageMeasurementsUtilities model's sizes getter
+ *
+ * Get the sizes of the vectors and matrices used by the solver to simulate
+ * ModelVoltageMeasurementsUtilities instance. Used by @p ModelMulti to generate right size matrices
+ * and vector for the solver.
+ */
+void
+ModelVoltageMeasurementsUtilities::getSize() {
+  sizeF_ = 0;  // No dynamics
+  sizeY_ = nbConnectedInputs_;  // All voltage inputs and their boolean activity values
+  sizeZ_ = nbConnectedInputs_;
+  sizeG_ = 0;
+  sizeMode_ = 1;
+
+  calculatedVars_.assign(nbCalculatedVars_, 0);
+}
+
+void
+ModelVoltageMeasurementsUtilities::evalF(double /*t*/, propertyF_t /*type*/) {
+  // No evalF function needed
+}
+
+void
+ModelVoltageMeasurementsUtilities::evalG(const double /*t*/) {
+  // No root function for this model
+}
+
+void
+ModelVoltageMeasurementsUtilities::evalJt(const double /*t*/, const double /*cj*/, SparseMatrix& /*jt*/, const int /*rowOffset*/) {
+  // No evalJt function needed
+}
+
+void
+ModelVoltageMeasurementsUtilities::evalJtPrim(const double /*t*/, const double /*cj*/, SparseMatrix& /*jt*/, const int /*rowOffset*/) {
+  // No evalJtPrim function needed
+}
+
+void
+ModelVoltageMeasurementsUtilities::evalZ(const double /*t*/) {
+  // No evalZ function needed
+}
+
+void
+ModelVoltageMeasurementsUtilities::collectSilentZ(BitMask* /*silentZTable*/) {
+  // Not needed here.
```

Comment:
I think you cann add something like: 
for (unsigned int s = 0; s < sizeZ_; ++s) {
      silentZTable[s].setFlags(NotUsedInContinuousEquations);
    }
@rosiereflo do u confirm? thx